core.coffee

```diff
@@ -434,6 +447,35 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				data = utils.cloneDeep(data)
+
+				buildReq = (req) =>
+					# This is a normal request
+					if req.url?
+						req['Content-Transfer-Encoding'] = 'binary'
+						if req.body?
+							req['Content-Type'] = 'application/json'
+							b = JSON.stringify(req.body)
+						else
+							req['Content-Type'] = 'application/http'
+							b = ''
+						req.body = "#{req.method} #{@compile(req)} HTTP/1.1\r\n#{b}"
+						return utils.omit(req, 'url', 'method')
+					# This is a changeset
+					else
+						if !utils.isArray(req) then throw new Error('ChangeSet must be passed as an array')
+						cs = mimemessage.factory(
+							contentType: 'multipart/mixed'
+							body: utils.map req, (b) ->
```

Comment:
`map` is not a function we require of the `utils` object - you can should be able to replace this with a coffeescript for loop though